scipy/stats/_multivariate.py

```diff
@@ -2979,25 +2963,23 @@ def _rvs(self, n, shape, dim, df, C, random_state):
         """
         random_state = self._get_random_state(random_state)
         # Get random draws A such that A ~ W(df, I)
-        A = super()._standard_rvs(n, shape, dim, df, random_state)
+        A = self._inv_standard_rvs(n, shape, dim, df, random_state)
 
         # Calculate SA = (CA)'^{-1} (CA)^{-1} ~ iW(df, scale)
-        eye = np.eye(dim)
         trtrs = get_lapack_funcs(('trtrs'), (A,))
 
         for index in np.ndindex(A.shape[:-2]):
             # Calculate CA
-            CA = np.dot(C, A[index])
-            # Get (C A)^{-1} via triangular solver
+            # Get CA = (C A^{-1}).T via triangular solver
             if dim > 1:
-                CA, info = trtrs(CA, eye, lower=True)
+                CA, info = trtrs(A[index], C.T, lower=True, trans=True)
```

Comment:
`trtrs` is basically `trsm` with diagonal zero check but since `cholesky` succeeded we don't need `trtrs` here and trsm would be sufficient.

Reply:
Ah yes, even better, in this case the matrix required to be triangular is `A[index]`, which is a valid Cholesky factor by construction.